### ISSUE TITLE
Heavy audio distortion after 15~ seconds of streaming

### ISSUE THREAD

**Issue by MrSpongen** [irtimmer/moonlight-embedded#497]
Please provide the following info.

**_NVidia Geforce Experience version:** 3.6.0.74
**Moonlight Embedded version:** 2.4.0
**Moonlight Embedded running on:** Raspberry Pi 2 B+

**Output of Moonlight Embedded:** Stream is working, a bit choppy, but after about 15 seconds, audio goes to full distortion and it's not bearable. Not sure what to answer here...

**What is the expected result?** Audio working.

**What happens instead of that?** Audio, even if I am in the menu of Steam big picture or launch directly into a game, it starts distorting about 15 seconds after stream starts. 

I receive errors of OOS Audio data (expected "number" but received "number"). Also getting some "Discarding RTP packet queued for too long" error. The audio isn't delayed or anything, it just gets extremely distorted after about 15-20 seconds. 

Running it from my desktop to my Pi, tried with default settings, 720@30, 1060@60, bitrates and all. Same issue with everything. Re-installed NOOBS and Raspbian from scratch and updated it all to the latest possible versions. I can record a clip and append here what is happening if that'll help.

My aplay, if it helps: 

pi@raspberrypi:~ $ aplay -l
**** List of PLAYBACK Hardware Devices ****
card 0: ALSA [bcm2835 ALSA], device 0: bcm2835 ALSA [bcm2835 ALSA]
  Subdevices: 8/8
  Subdevice #0: subdevice #0
  Subdevice #1: subdevice #1
  Subdevice #2: subdevice #2
  Subdevice #3: subdevice #3
  Subdevice #4: subdevice #4
  Subdevice #5: subdevice #5
  Subdevice #6: subdevice #6
  Subdevice #7: subdevice #7
card 0: ALSA [bcm2835 ALSA], device 1: bcm2835 ALSA [bcm2835 IEC958/HDMI]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
pi@raspberrypi:~ $ aplay -L
null
    Discard all samples (playback) or generate zero samples (capture)
pulse
    PulseAudio Sound Server
default:CARD=ALSA
    bcm2835 ALSA, bcm2835 ALSA
    Default Audio Device
sysdefault:CARD=ALSA
    bcm2835 ALSA, bcm2835 ALSA
    Default Audio Device
dmix:CARD=ALSA,DEV=0
    bcm2835 ALSA, bcm2835 ALSA
    Direct sample mixing device
dmix:CARD=ALSA,DEV=1
    bcm2835 ALSA, bcm2835 IEC958/HDMI
    Direct sample mixing device
dsnoop:CARD=ALSA,DEV=0
    bcm2835 ALSA, bcm2835 ALSA
    Direct sample snooping device
dsnoop:CARD=ALSA,DEV=1
    bcm2835 ALSA, bcm2835 IEC958/HDMI
    Direct sample snooping device
hw:CARD=ALSA,DEV=0
    bcm2835 ALSA, bcm2835 ALSA
    Direct hardware device without any conversions
hw:CARD=ALSA,DEV=1
    bcm2835 ALSA, bcm2835 IEC958/HDMI
    Direct hardware device without any conversions
plughw:CARD=ALSA,DEV=0
    bcm2835 ALSA, bcm2835 ALSA
    Hardware device with all software conversions
plughw:CARD=ALSA,DEV=1
    bcm2835 ALSA, bcm2835 IEC958/HDMI
    Hardware device with all software conversions
pi@raspberrypi:~ $

Put up a video of the issue here: https://www.youtube.com/watch?v=oIZMszpcohE

**Comment by irtimmer** [comment]
Have you tried to run Moonlight Embedded outside of a desktop environment? With the Raspberry Pi 1 and an earlier version of Moonlight Embedded this was a real problem.

**Comment by MrSpongen** [comment]
Yes, I tried that as well. It seemed to work slightly better, but the audio was still being distorted after a while. Kept getting the OOS audio data unfortunately. Perhaps my Raspberry Pi 2 B+ can't cope with Moonlight embedded? I got it working like a year ago, but wanted to reformat it and this time this issue kept appearing. I didn't use a wireless dongle back then though. Are there any reports on using one creating a lot of demand out of the Pi or something?

**Comment by irtimmer** [comment]
Wireless connections can be very problematic because of packet loss. But that should also result in video corruption. By default Raspbian provides a slow audio decoder (with slightly better sound). So you could try to install http://archive.itimmer.nl/raspbian/libopus0_1.1.2-1_armhf.deb to check if it improves the situation as it would lower the cpu usage. You could also try to use another power adapter as the Raspberry Pi could lower its performance when it hasn't enough power (although I thought you should get a visual indication if this happens)

**Comment by MrSpongen** [comment]
My bad. Poor explanation skills showing here. I meant I didn't use a wireless dongle for my Xbox 360 controller, my ethernet cable was plugged in. Sorry for the bad explanation! What I was thinking is that was probably the only "big" change I've done since I tried Limelight as it was known then. Last time I tried streaming, there wasn't this neat embedded solution :)

Also, I got myself one of those specially crafted power adapters for the Raspberry Pi when I got a newer model, when I went from the first to the second. So hopefully my power shouldn't be an issue. 

I'll try the .deb-package.
